src/newcharacter.cpp

```diff
@@ -1088,8 +1070,60 @@ int set_profession(WINDOW *w, player *u, int &points)
     ctxt.register_action("NEXT_TAB");
     ctxt.register_action("SORT");
     ctxt.register_action("HELP_KEYBINDINGS");
+    ctxt.register_action("FILTER");
+
+    bool recalc_profs = true;
+    int profs_length;
+    std::string filterstring = "";
```

Comment:
You probably copied that from somewhere, but strings don't need to be initialized to an empty string. They automatically initialize themself. `std::string filterstring;` is enough.